### ISSUE TITLE
Pod fails to download image from private registry

### ISSUE THREAD

**Issue by davesargrad** [kubernetes/kubernetes#85904]
Guys,
I have a cluster with 2 nodes.

![image](https://user-images.githubusercontent.com/48793689/70149542-523e7b80-1676-11ea-9e70-03fd24bac1d0.png)

My pod is running on spain. As you see in the following the camel-k-kit is in error. 
![image](https://user-images.githubusercontent.com/48793689/70149562-62565b00-1676-11ea-8448-d1e9aecb3954.png)

The reason for the error is a failure to download an image from my local insecure docker registry.

![image](https://user-images.githubusercontent.com/48793689/70149662-99c50780-1676-11ea-97d8-fefc51a8e3c4.png)

You can see the image in my catalog.
![image](https://user-images.githubusercontent.com/48793689/70149686-a6496000-1676-11ea-84dc-79d178e158b2.png)

![image](https://user-images.githubusercontent.com/48793689/70155754-d433a200-1680-11ea-9d2d-e4fec3b55bf7.png)

and from spain (the k8s worker running the pod)
![image](https://user-images.githubusercontent.com/48793689/70152890-01318600-167c-11ea-9dee-a45d0618a623.png)

I've actually had this working on another k8s cluster, in pretty much exactly the same configuration (I think).

I am behind a corporate proxy, so I have to setup various files.

/etc/systemd/system/docker.service.d/http-proxy.conf
![image](https://user-images.githubusercontent.com/48793689/70149774-cda02d00-1676-11ea-864f-c31f278083dd.png)

I also configure ~root/.docker/config.json on spain (k8s worker) and benin (k8s master)
![image](https://user-images.githubusercontent.com/48793689/70149881-ffb18f00-1676-11ea-9b80-2fff5f7e6c2b.png)

How do I diagnose the failure to download the image from my private docker instance running on 10.93.97.123:5007?

/sig architecture

**Comment by davesargrad** [comment]
I've marked this as an architecture "sig" only because I've struggled with k8s and proxy in the past.  In the end I've always gotten past the proxy issues.

I've seen enough struggles that others have also had with this (through a look at the issues on github, and the various posts on stackoverflow). I'd think that by this time there is a clean way to diagnose the specific problem and documentation on the solution. It seems that every time I bring up another k8s cluster (even if i follow the procedures that I've finally gotten working before), I end up with something that relates to proxy.

In reality I think this is likely a simple fix, yet it costs hours of struggle. I think once there is solid documentation and guidance that walks people through isolating such proxy related issues, the user experience with k8s will increase dramatically.

Anyways, I'd greatly appreciate guidance on how to isolate and fix this. I will do my best to follow up with details on this issue, that hopefully will help others.
